### PR TITLE
Remove CUDA-specific reactions code

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

File: SourceCpp/React.cpp [hunk 1]
```diff
@@ -146,30 +146,26 @@ PeleC::react_state(
...
+          amrex::Vector<amrex::Real> h_rY_src_in(ncells * NUM_SPECIES);
+          amrex::Gpu::DeviceVector<amrex::Real> rY_src_in(ncells * NUM_SPECIES);
+          amrex::Real* d_rY_src_in = rY_src_in.data();
+          amrex::Vector<amrex::Real> h_re_in(ncells);
+          amrex::Gpu::DeviceVector<amrex::Real> re_in(ncells);
+          amrex::Real* d_re_in = re_in.data();
+          amrex::Vector<amrex::Real> h_re_src_in(ncells);
+          amrex::Gpu::DeviceVector<amrex::Real> re_src_in(ncells);
+          amrex::Real* d_re_src_in = re_src_in.data();
+
```
Comment (hsitaram) [inline SourceCpp/React.cpp:162]: do we have to clear these at the end like h_rY_in.clear()?

Comment (jrood-nrel) [reply SourceCpp/React.cpp:162]: This is a good thought, but I think when they fall out of scope, their destructor does it. I will double check.

Comment (jrood-nrel) [reply SourceCpp/React.cpp:162]: I can't find any example of a `clear` happening in AMR-Wind and they don't mention it here https://amrex-codes.github.io/amrex/docs_html/GPU.html#gpu-vectors so I'm assuming all these vectors get cleared automatically even on device.

Comment (hsitaram) [reply SourceCpp/React.cpp:162]: I was hoping that will be the case.. you never know sometimes!

**Review by hsitaram — APPROVED** [review]

File: SourceCpp/React.cpp (hunk 1 above)
Comment (jrood-nrel) [reply SourceCpp/React.cpp:162]: Definitely. I will still need to do more testing of this by running longer simulations once it's merged.

**Pull request merged**